unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarMoreMenu.cs

```diff
@@ -68,18 +73,30 @@ public virtual void Initialize(TaskbarHUDView view)
             view.moreButton.SetToggleState(false);
         });
 
+        toggleAvatarNamesButton.mainButton.onClick.AddListener(() => { avatarNamesVisible.Set(!avatarNamesVisible.Get()); });
+        OnAvatarNamesVisibleChanged(avatarNamesVisible.Get(), false);
+
         collapseBarButton.mainButton.onClick.AddListener(ToggleCollapseBar);
 
         hideUIButton.mainButton.onClick.AddListener(ToggleHideUI);
 
         tutorialButton.mainButton.onClick.AddListener(() => { OnRestartTutorial?.Invoke(); });
     }
 
+    private void OnAvatarNamesVisibleChanged(bool current, bool previous)
+    {
+        if (current)
+            toggleAvatarNamesButton.buttonText.SetText("Hide Names");
+        else
+            toggleAvatarNamesButton.buttonText.SetText("Show Names");
```

Comment:
@Suduck fixed the issue with the default settings. Make sure to test in a clean environment since your current one already have the name opacity saved at 0.